pkg/apis/core/v1/helper/helpers.go

```diff
@@ -246,6 +248,44 @@ func NodeSelectorRequirementsAsSelector(nsm []v1.NodeSelectorRequirement) (label
 	return selector, nil
 }
 
+// MatchNodeSelectorTermsORed checks whether the node labels and fields match node selector terms in ORed;
+// nil or empty term matches no objects.
+func MatchNodeSelectorTermsORed(
+	nodeSelectorTerms []v1.NodeSelectorTerm,
+	nodeLabels labels.Set,
+	nodeFields labels.Set,
+) bool {
+	for _, req := range nodeSelectorTerms {
+		labelSelector, err := NodeSelectorRequirementsAsSelector(req.MatchExpressions)
+		if err != nil {
+			glog.V(10).Infof("Failed to parse MatchExpressions: %+v,regarding as not match.", req.MatchExpressions)
+			return false
+		}
+
+		if labelSelector.Selectable() && !labelSelector.Matches(nodeLabels) {
+			continue
+		}
+
+		fieldSelector, err := NodeSelectorRequirementsAsSelector(req.MatchFields)
+		if err != nil {
+			glog.V(10).Infof("Failed to parse MatcheFields: %+v,regarding as not match.", req.MatchFields)
+			return false
```

Comment:
should this return or continue? Other valid terms could still match

Reply:
`continue` is better :)